src/service-api.ts

```diff
@@ -183,6 +185,92 @@ const plugin: FastifyPluginAsync<GraaspImportZipPluginOptions> = async (fastify,
     },
   );
+
+  // download item as zip
+  fastify.get<{ Params: { itemId: string } }>(
+    '/zip-export/:itemId',
+    //{ schema: zipExport },
+    async ({ member, params: { itemId }, log }, reply) => {
+      const addFileToZip = async (item: Item, dirPath) => {
+        // get item and its related data
+        const itemExtra = item.extra as Extra;
+
+        switch (item.type) {
+          case SERVICE_ITEM_TYPE:
+            // TODO: add uploaded files into zip
+            break;
+          case ItemType.DOCUMENT:
+            archive.append(itemExtra.document?.content, { name: `${dirPath}${item.name}.graasp` });
+            break;
+          // TODO: better handle the case of link / app
+          case ItemType.LINK:
+            archive.append(`_source\n${itemExtra.embeddedLink?.url}\nlinkType: 0\n`, {
+              name: `${dirPath}${item.name}.url`,
+            });
+            break;
+          case ItemType.APP:
+            archive.append(`_source\n${itemExtra.app?.url}\nlinkType: 1\n`, {
+              name: `${dirPath}${item.name}.url`,
+            });
+            break;
+          case ItemType.FOLDER:
+            // append description
+            if (dirPath === '') {
+              dirPath = `/${item.name}/`;
+            } else {
+              dirPath = `${dirPath}${item.name}/`;
+            }
+            if (item.description) {
+              archive.append(item.description, { name: `${dirPath}${item.name}.description.html` });
+            }
+            // eslint-disable-next-line no-case-declarations
+            const tasks = iTM.createGetChildrenTaskSequence(member, item.id, true);
+            // eslint-disable-next-line no-case-declarations
+            const subItems = await runner.runSingleSequence(tasks, log);
+            (subItems as Item[]).forEach((subItem) => {
+              addFileToZip(subItem, dirPath);
+            });
+        }
+      };
+      const getItemTask = iTM.createGetTask(member, itemId);
+      const item = await runner.runSingle(getItemTask);
+      const folderName = item.name;
+      const folderPath = `${__dirname}/${folderName}.zip`;
+
+      const archive = archiver.create('zip', { store: true });
+      const output = fs.createWriteStream(`${folderName}.zip`);
+
+      // initialize archiver
+      output.on('close', function () {
+        log.debug(archive.pointer() + ' total bytes');
+        log.debug('archiver has been finalized and the output file descriptor has closed.');
+      });
+      output.on('end', function () {
+        log.debug('Data has been drained');
+      });
+      archive.on('warning', function (err) {
+        if (err.code === 'ENOENT') {
+          log.debug(err);
+        } else {
+          throw err;
+        }
+      });
+      archive.on('error', function (err) {
+        throw err;
+      });
+      // pipe archive data to the file
+      archive.pipe(output);
+
+      const rootPath = '';
+      await addFileToZip(item, rootPath);
```

Comment:
It's where you save the archive on the server isn't it? most probably we want to factor out in `constants` and save it in a `tmp` folder.